sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -67,6 +73,359 @@ describe(
+
+      // Assert
+      expect(result).toStrictEqual([
+        "Process finalized with success.",
+        "Generated file: none",
+        "Uploaded records: 0",
+      ]);
+      const studentNotificationCount = await db.notification.count({
+        where: {
+          notificationMessage: {
+            id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+          },
+          metadata: {
+            disbursementId:
+              application.currentAssessment.disbursementSchedules[0].id,
+          },
+          user: { id: application.student.user.id },
+          creator: { id: systemUsersService.systemUser.id },
+        },
+      });
+      expect(studentNotificationCount).toBe(3);
+    });
+
+    it("Should not create a notification for the student for a disbursement when an attempt is made to create the 2nd notification before 7 days from the first notification.", async () => {
+      // Arrange
+      // Student with invalid SIN to block the disbursement.
+      const student = await saveFakeStudent(db.dataSource, undefined, {
+        sinValidationInitialValue: {
+          isValidSIN: false,
+        },
+      });
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.Signed,
+            msfaaInitialValues: {
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+      // Student application.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student, msfaaNumber },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+      const todaysDate = new Date();
+      // Create 1 pre-existing notification for the above created disbursement.
+      await saveFakeNotification(
+        db.dataSource,
+        {
+          user: student.user,
+          creator: systemUsersService.systemUser,
+          notificationMessage: {
+            id: NotificationMessageType.StudentNotificationDisbursementBlocked,
+          } as NotificationMessage,
+        },
+        {
+          initialValue: {
+            messagePayload: {
+              email_address: student.user.email,
+              template_id:
+                NotificationMessageType.StudentNotificationDisbursementBlocked,
+              personalisation: {
+                givenNames: student.user.firstName ?? "",
+                lastName: student.user.lastName,
+              },
+            },
+            metadata: {
+              disbursementId:
+                application.currentAssessment.disbursementSchedules[0].id,
+            },
+            createdAt: new Date(todaysDate.setDate(todaysDate.getDate() - 6)),
```

Comment:
We have a utility addDays.

![image](https://github.com/bcgov/SIMS/assets/54600590/e534f169-73c4-4423-90f8-4267323e7782)